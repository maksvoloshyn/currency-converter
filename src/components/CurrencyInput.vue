<template>
    <input
        class="border px-3 py-2 rounded-lg"
        :value="inputValue"
        @input="updateValue($event.target.value)"
        @focus="toggleActiveState(true)"
        @blur="toggleActiveState(false)"
    />
</template>

<script>
    import {localiseExchangeValue} from '@/utils';

    export default {
        name: 'currency-input',

        props: {
            amount: {
                type: Number,
                default: 0,
            },
        },

        data() {
            return {
                isInputActive: false,
            };
        },

        computed: {
            inputValue() {
                return this.isInputActive ? this.$el.value : localiseExchangeValue(this.amount);
            },
        },

        methods: {
            updateValue(value) {
                this.$emit('change', parseFloat(value));
            },

            toggleActiveState(state) {
                this.isInputActive = state;
            },
        },
    };
</script>
