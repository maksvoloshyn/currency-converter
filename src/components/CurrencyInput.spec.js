import {shallowMount} from '@vue/test-utils';
import CurrencyInput from '@/components/CurrencyInput';

function factory() {
    return shallowMount(CurrencyInput, {
        propsData: {
            amount: 12,
        },
    });
}

describe('CurrencyInput', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = factory();
    });

    it('renders correct template', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    describe('when input element emits `input` event', () => {
        it('emits `change` event with a given value', async () => {
            await wrapper.find('input').setValue('15');

            expect(wrapper.emitted('change')[0][0]).toBe(15);
        });
    });
});
