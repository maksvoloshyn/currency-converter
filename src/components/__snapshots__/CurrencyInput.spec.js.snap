// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyInput renders correct template 1`] = `
<input
  class="border px-3 py-2 rounded-lg"
/>
`;
